crashed when resource does not exist

### Terraform Version
Terraform v0.9.0 (MacOs Sierra)

### Affected Resource(s)
When you try to reference a resource which is actually generated by a module and does not exist you get this crashed, however under 0.8.6 you can get the exact details about which resource has not been created. 

### Panic Output
```
goroutine 432 [running]:
github.com/hashicorp/terraform/config/module.(*Tree).Validate(0xc42039b180, 0xc42042f170, 0x8)
	/opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:358 +0xe65
github.com/hashicorp/terraform/terraform.(*Context).Validate(0xc420432360, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/context.go:644 +0xd0
github.com/hashicorp/terraform/backend/local.(*Local).context(0xc4200a70e0, 0xc4200b4630, 0x2, 0x2, 0x3, 0xc420130440, 0x2)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend_local.go:82 +0x262
github.com/hashicorp/terraform/backend/local.(*Local).opPlan(0xc4200a70e0, 0x6ff44a0, 0xc420084260, 0xc4200b4630, 0xc4203abe30)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend_plan.go:57 +0x22f
github.com/hashicorp/terraform/backend/local.(*Local).(github.com/hashicorp/terraform/backend/local.opPlan)-fm(0x6ff44a0, 0xc420084260, 0xc4200b4630, 0xc4203abe30)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend.go:224 +0x52
github.com/hashicorp/terraform/backend/local.(*Local).Operation.func1(0xc4200a70e0, 0xc42042eba0, 0xc42042eb90, 0x6ff44a0, 0xc420084260, 0xc4200b4630, 0xc4203abe30)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend.go:246 +0x9a
created by github.com/hashicorp/terraform/backend/local.(*Local).Operation
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend.go:247 +0x192
```

### Expected Behavior
Terraform should tell you which resource does not exist (or give you a hint such as in 0.8.6)

### Actual Behavior
Crashed

### Steps to Reproduce
execute terraform plan

Sure, sorry about that I can't use same code but I was able to reproduce this issue on this small example:

example1.tf

```
resource "aws_instance" "web" {
  ami           = "${data.aws_ami.ubuntu.id}"
  instance_type = "t2.micro"

  security_groups = ["${module.my_security_groups.my_sg.id}"]

  tags {
    Name = "HelloWorld"
  }
}
```

output
```
panic: module not found in children: my_security_groups

goroutine 420 [running]:
github.com/hashicorp/terraform/config/module.(*Tree).Validate(0xc420018960, 0xc4204010b0, 0x8)
	/opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:358 +0xe65
github.com/hashicorp/terraform/terraform.(*Context).Validate(0xc4204826c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/context.go:644 +0xd0
github.com/hashicorp/terraform/backend/local.(*Local).context(0xc42008eb40, 0xc420151b80, 0x2, 0x2, 0x3, 0xc420474a80, 0x2)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend_local.go:82 +0x262
github.com/hashicorp/terraform/backend/local.(*Local).opPlan(0xc42008eb40, 0x6ff44a0, 0xc420012560, 0xc420151b80, 0xc420479710)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend_plan.go:57 +0x22f
github.com/hashicorp/terraform/backend/local.(*Local).(github.com/hashicorp/terraform/backend/local.opPlan)-fm(0x6ff44a0, 0xc420012560, 0xc420151b80, 0xc420479710)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend.go:224 +0x52
github.com/hashicorp/terraform/backend/local.(*Local).Operation.func1(0xc42008eb40, 0xc420400ae0, 0xc420400ad0, 0x6ff44a0, 0xc420012560, 0xc420151b80, 0xc420479710)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend.go:246 +0x9a
created by github.com/hashicorp/terraform/backend/local.(*Local).Operation
	/opt/gopath/src/github.com/hashicorp/terraform/backend/local/backend.go:247 +0x192
```

--- a/config/module/test-fixtures/validate-module-unknown/main.tf
+++ b/config/module/test-fixtures/validate-module-unknown/main.tf
@@ -1,0 +1,3 @@
+resource "null_resource" "var" {
+    key = "${module.unknown.value}"
+}

--- a/config/module/tree.go
+++ b/config/module/tree.go
@@ -354,8 +354,10 @@ func (t *Tree) Validate() error {
 
 			tree, ok := children[mv.Name]
 			if !ok {
-				// This should never happen because Load watches us
-				panic("module not found in children: " + mv.Name)
+				newErr.Add(fmt.Errorf(
+					"%s: undefined module referenced %s",
+					source, mv.Name))
+				continue
 			}
 
 			found := false

--- a/config/module/tree_test.go
+++ b/config/module/tree_test.go
@@ -424,6 +424,18 @@ func TestTreeValidate_requiredChildVar(t *testing.T) {
 	}
 }
 
+func TestTreeValidate_unknownModule(t *testing.T) {
+	tree := NewTree("", testConfig(t, "validate-module-unknown"))
+
+	if err := tree.Load(testStorage(t), GetModeNone); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	if err := tree.Validate(); err == nil {
+		t.Fatal("should error")
+	}
+}
+
 const treeLoadStr = `
 root
   foo (path: foo)

